### ISSUE TITLE
Crash when created an excluded file

### ISSUE THREAD

**Issue by williamdes** [zed-industries/zed#10890]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I tried to create a `.gitignore` file

```
Thread "main" panicked with "called `Option::unwrap()` on a `None` value" at crates/gpui/src/platform/cosmic_text/text_system.rs:308:14
```

```json
  "file_scan_exclusions": [
...
    "**/.gitattributes",
    "**/.gitignore",
...
  ],
```

### Environment

Zed: v1.0.0 (Zed Dev 615de381dabd3501ff8863a201c2df1be1b70b46)
OS: Debian 12
Memory: 62.4 GiB
Architecture: x86_64

**Comment by apricotbucket28** [comment]
Hello! Is this still happening? If so, could you share the exact steps to reproduce the issue?

**Comment by williamdes** [comment]
> Hello! Is this still happening? If so, could you share the exact steps to reproduce the issue?

Yeah, got it this afternoon

Read the issue, you need to create a file that you excluded from the tree
Right click on folder -> new file and use a file name that is excluded

**Comment by apricotbucket28** [comment]
I cannot reproduce the issue, and it seems very unlikely that this is due to the files being excluded since the crash happens in the text system. Could you share your settings?

**Comment by williamdes** [comment]
> I cannot reproduce the issue, and it seems very unlikely that this is due to the files being excluded since the crash happens in the text system. Could you share your Zed settings? (mainly font settings and such) It'd also be helpful if you could record a video 🙂

It is totally related.

In fact I added a special name to file_scan_exclusions and tried to create the file and it crashed.
```json
"file_scan_exclusions": [
"**/.make-it-crash"
]
```

```
Thread "main" panicked with "called `Option::unwrap()` on a `None` value" at crates/gpui/src/platform/cosmic_text/text_system.rs:297:14
   0: Zed::reliability::init_panic_hook::{closure#0}
             at crates/zed/src/reliability.rs:54:29
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/boxed.rs:2034:9
      std::panicking::rust_panic_with_hook
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:783:13
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:649:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:171:18
   4: rust_begin_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
   5: core::panicking::panic_fmt
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
   6: core::panicking::panic
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:145:5
   7: core::option::unwrap_failed
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/option.rs:1985:5
   8: <core::option::Option<swash::scale::image::Image>>::unwrap
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/option.rs:933:21
      <gpui::platform::cosmic_text::text_system::CosmicTextSystemState>::raster_bounds
             at crates/gpui/src/platform/cosmic_text/text_system.rs:297:14
      <gpui::platform::cosmic_text::text_system::CosmicTextSystem as gpui::platform::PlatformTextSystem>::glyph_raster_bounds
             at crates/gpui/src/platform/cosmic_text/text_system.rs:165:9
```

**Comment by SomeoneToIgnore** [comment]
@williamdes , sounds like you have something else in your settings that makes it panic.

As it shows in the backtrace, the crash is related to font rendering, so listening to the previous message and providing your settings, especially

> mainly font settings and such

is a way forward.

Here's the latest Preview that does not crash (albeit the behavior is still not good):
https://github.com/zed-industries/zed/assets/2690773/036cc3c4-a26a-4a73-b9d8-21ab892c73e6

**Comment by SomeoneToIgnore** [comment]
Great, now we have to ask you, for the 3rd time in this issue, to provide your settings.

**Comment by williamdes** [comment]
> Great, now we have to ask you, for the 3rd time in this issue, to provide your settings.

Gosh, sorry about that 🤦🏻 
Well, I already provided them.
That said, to ensure no other setting was interfering with this I nuked all my settings and kept
```json

{
  "file_scan_exclusions": [
    "**/.gitignore"
  ]
}
```

The bug does not change

**Comment by SomeoneToIgnore** [comment]
Yes, I can read and noticed that settings excerpt from the very beginning.
The whole point is to get ALL your settings, as it's definitely panicking on something text-related.

> mainly font settings and such

at least those are needed — I suspect, you have a custom font, maybe a custom font size, maybe something else (ergo asking for ALL settings)

**Comment by SomeoneToIgnore** [comment]
Ohh, I see, sorry for misreading the last comment — so there's something system-related then?

I will try to poke at it, but it's quite hard to understand what's not matching the "regular" experience.
This is quite hard to fix with such little info though, so I might have to come back and ask you to try a build/patch, since I'm totally blind here.

**Comment by williamdes** [comment]
> Ohh, I see, sorry for misreading the last comment — so there's something system-related then?

I am using XFCE 4.18 on  Debian 12 and did not change font stuff.

> I will try to poke at it, but it's quite hard to understand what's not matching the "regular" experience.
This is quite hard to fix with such little info though, so I might have to come back and ask you to try a build/patch, since I'm totally blind here.

Yes, sure, I will test it right away

On your video it looked like there is a GUI bug, maybe fix this first and we will see if the crash persists

**Comment by SomeoneToIgnore** [comment]
There was more than one bug related to the directory creation, and overall the ergonomics required some improvements, so it was a relatively long one: https://github.com/zed-industries/zed/pull/12620

The fix is coming this Wednesday with Zed 0.139

**Comment by williamdes** [comment]
Thank you for the fix !
I downloaded the nightly build and it works 🎉
